src/config/unit_test.rs

```diff
@@ -109,11 +109,34 @@ fn walk(
     let mut results = Vec::new();
     let mut targets = Vec::new();
 
-    if let Some(target) = transforms.get_mut(node) {
-        for input in inputs.clone() {
-            target.transform.transform_into(&mut results, input);
+    // Use `remove` to take ownership.
+    if let Some((key, mut target)) = transforms.remove_entry(node) {
+        match target.transform {
+            Transform::Function(ref mut t) => {
+                for input in inputs.clone() {
+                    t.transform(&mut results, input)
+                }
+                targets = target.next.clone();
+                transforms.insert(key, target);
+            }
+            Transform::Task(t) => {
+                error!("Using a recently refactored `TaskTransform` in a unit test. You may experience limited support for multiple inputs.");
```

Comment:
It looks like our `reduce` behavior tests bump into the problem of consumption of the transform here. @lukesteensen would you advise disabling these behavior tests or should this ticket also refactor those?

Reply:
Do you have an idea for how to handle it? It'd be nice to include if it's not too big an addition, but otherwise I'm fine to open an issue and reenable in a follow up PR.

Reply:
Let me give it a shot today.

Reply:
Resolved!